ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/BufferedTransactionsReader.scala

```diff
@@ -65,7 +72,13 @@ private[events] class BufferedTransactionsReader(
       verbose: Boolean,
   )(implicit
       loggingContext: LoggingContext
-  ): Source[(Offset, GetTransactionTreesResponse), NotUsed] =
+  ): Source[(Offset, GetTransactionTreesResponse), NotUsed] = {
+    val span =
```

Comment:
Have you got a way of testing it? I expect that we will end up with two unrelated spans one for BufferedTransactionsReader and one for inner TransactionsReader. It would be better to have one become a child of the other

Reply:
My intuition was that this would work as well with nested traces. But I admit I have never played with traces and Jaeger. Followed your advice and created a span over the source from the buffer. I'll test it nevertheless

Reply:
Nicely done!